server/channels/api4/channel_category.go

```diff
@@ -35,7 +35,10 @@ func getCategoriesForTeamForUser(c *Context, w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	w.Write(categoriesJSON)
+	if _, err := w.Write(categoriesJSON); err != nil {
+		c.Err = model.NewAppError("getCategoriesForTeamForUser", "api.write_error", nil, err.Error(), http.StatusInternalServerError).Wrap(err)
+		return
+	}
```

Comment:
If the write fails, there is no point is sending another error to the client. It will likely also fail. Let's just log an error and be  done. You can follow the pattern from 
https://github.com/mattermost/mattermost/blob/f1903754034b395b1fb0aef337ed689b17e3d61a/server/channels/api4/system.go#L816-L818

Reply:
Okay, thanks - makes sense. Will fix!